lampod/src/actions/handler.rs

```diff
@@ -150,13 +150,26 @@ impl Handler for LampoHandler {
                     channel_type,
                 }));
                 Ok(())
-            }
+            },
             ldk::events::Event::ChannelClosed {
                 channel_id,
                 user_channel_id,
                 reason,
+                counterparty_node_id,
+                channel_funding_txo,
                 ..
             } => {
+                let node_id = if let Some(id) = counterparty_node_id {
+                    id.to_string()
+                } else {
+                    "Node_id not found".to_string()
+                };
```

Comment:
This is wrong, you are setting the `Node_id not found`! if there is no node id you should keep it optional and pass it down as an optional value